src/ess/reduce/widgets/_optional_widget.py

```diff
@@ -0,0 +1,66 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+from typing import Any
+
+from ipywidgets import HTML, HBox, Layout, RadioButtons, Widget
+
+from ._config import default_style
+
+
+class OptionalWidget(HBox):
+    """Wrapper widget to handle optional widgets.
+
+    When you retrieve the value of this widget,
+    it will return the value of the wrapped widget.
+    The parameter should be set as ``None`` or the actual value.
+    """
+
+    def __init__(self, wrapped: Widget, name: str = '', **kwargs) -> None:
+        self.name = name
+        self.wrapped = wrapped
+        self._option_box = RadioButtons(
```

Comment:
Maybe it's simpler to use a single checkbox here? That would let us remove the layout code, since we would not have to manually place the two radio elements on the same line.

Reply:
I'm thinking something like:

```python
import ipywidgets as ipw

c = ipw.Checkbox(description='')
s = ipw.Stack([ipw.Label('None'), ipw.Text()], selected_index=0)
ipw.dlink((c, 'value'), (s, 'selected_index'))
ipw.HBox([c, s])
```

![Screenshot from 2024-08-13 12-49-20](https://github.com/user-attachments/assets/1dc04852-4ea6-47b8-804e-fe425ffcfaff)
![Screenshot from 2024-08-13 12-49-31](https://github.com/user-attachments/assets/ca6c4f8d-80d9-4144-8cf3-8012152a3b91)

Reply:
Or, same but using `.disabled` like you did in this PR.

Reply:
But it doesn't matter much. If you think it's better like it is just go with that.